src/components/structures/auth/ForgotPassword.js

```diff
@@ -175,95 +203,144 @@ module.exports = React.createClass({
+        if (!SdkConfig.get()['disable_custom_urls']) {
+            editLink = <a className="mx_AuthBody_editServerDetails"
+                href="#" onClick={this.onEditServerDetailsClick}
+            >
+                {_t('Change')}
+            </a>;
+        }
+
+        return <div>
+            <h3>
+                {yourMatrixAccountText}
+                {editLink}
+            </h3>
+            {errorText}
+            <form onSubmit={this.onSubmitForm}>
+                <div className="mx_AuthBody_fieldRow">
+                    <input className="mx_Login_field" ref="user" type="text"
+                        name="reset_email" // define a name so browser's password autofill gets less confused
+                        value={this.state.email}
+                        onChange={this.onInputChanged.bind(this, "email")}
+                        placeholder={_t('Email')} autoFocus />
+                </div>
+                <div className="mx_AuthBody_fieldRow">
+                    <input className="mx_Login_field" ref="pass" type="password"
+                        name="reset_password"
+                        value={this.state.password}
+                        onChange={this.onInputChanged.bind(this, "password")}
+                        placeholder={_t('Password')} />
+                    <input className="mx_Login_field" ref="pass" type="password"
+                        name="reset_password_confirm"
+                        value={this.state.password2}
+                        onChange={this.onInputChanged.bind(this, "password2")}
+                        placeholder={_t('Confirm')} />
+                </div>
+                <span>{_t(
+                    'A verification email will be sent to your inbox to confirm ' +
+                    'setting your new password.',
+                )}</span>
+                <input className="mx_Login_submit" type="submit" value={_t('Send Reset Email')} />
+            </form>
+            <a className="mx_AuthBody_changeFlow" onClick={this.onLoginClick} href="#">
+                {_t('Sign in instead')}
+            </a>
+        </div>;
+    },
+
+    renderSendingEmail() {
+        const Spinner = sdk.getComponent("elements.Spinner");
+        return <Spinner />;
+    },
+
+    renderEmailSent() {
+        return <div>
+            {_t("An email has been sent to %(emailAddress)s. Once you've followed the " +
+                "link it contains, click below.", { emailAddress: this.state.email })}
+            <br />
+            <input className="mx_Login_submit" type="button" onClick={this.onVerify}
+                value={_t('I have verified my email address')} />
+        </div>;
+    },
+
+    renderDone() {
+        return <div>
+            <p>{_t('Your password has been reset')}.</p>
+            <p>{_t('You have been logged out of all devices and will no longer receive push notifications. ' +
+                'To re-enable notifications, sign in again on each device')}.</p>
```

Comment:
would be nice to have this either be indented to line up with the other line (similar to how line 302 deals with this) or to have them on independent lines (similar to 284).

Reply:
Ah yeah, still not sure I've settled on my favorite style for this... Also, most of this code is moved. Anyway, I'll update this bit.

Reply:
These lines also did a strange thing of leaving the last full stop outside the string... 😕 Fixed now.